flink-connectors/flink-connector-cassandra/src/test/java/org/apache/flink/streaming/connectors/cassandra/CassandraConnectorITCase.java

```diff
@@ -470,6 +477,41 @@ public void testCassandraTableSink() throws Exception {
 		Assert.assertTrue("The input data was not completely written to Cassandra", input.isEmpty());
 	}
 
+	@Test
+	public void testCassandraBatchPojoFormat() throws Exception {
+
+		session.execute(CREATE_TABLE_QUERY.replace(TABLE_NAME_VARIABLE, "batches"));
+
+		CassandraPojoSink<CustomCassandraAnnotatedPojo> sink = new CassandraPojoSink<>(CustomCassandraAnnotatedPojo.class, builder);
+		sink.open(new Configuration());
+
+		List<CustomCassandraAnnotatedPojo> customCassandraAnnotatedPojos = IntStream.range(0, 20)
+			.mapToObj(x -> new CustomCassandraAnnotatedPojo(UUID.randomUUID().toString(), x, 0))
+			.collect(Collectors.toList());
+
+		customCassandraAnnotatedPojos.forEach(sink::send);
+		sink.close();
+
+		InputFormat<CustomCassandraAnnotatedPojo, InputSplit> source = new CassandraPojoInputFormat<>(SELECT_DATA_QUERY.replace(TABLE_NAME_VARIABLE, "batches"), builder, CustomCassandraAnnotatedPojo.class);
```

Comment:
let's add a sanity check that the table actually contains 20 records to guard against errors during the writing.

Reply:
ok, it may be a little paranoid nop ? :)